spec/01-unit/04-prefix_handler_spec.lua

```diff
@@ -768,16 +768,75 @@ describe("NGINX conf compiler", function()
 
       lazy_teardown(function() cleanup() end)
 
-      it("injects the wasm{} subsystem", function()
-        local conf = assert(conf_loader(nil, {
-          wasm = "on",
-          wasm_filters_path = temp_dir,
-        }))
-        assert.equal(conf.wasm_filters_path, temp_dir)
-
-        local nginx_conf = prefix_handler.compile_nginx_conf(conf)
-        assert.matches("wasm {", nginx_conf)
-        assert.matches("module empty-filter " .. filter .. ";", nginx_conf, nil, true)
+      local _compile = function(cfg, config_compiler, debug)
+        local ngx_conf = config_compiler(assert(conf_loader(nil, cfg)))
+        if debug then
+          print(ngx_conf)
+        end
+        return ngx_conf
+      end
+      local ngx_cfg = function(cfg, debug) return _compile(cfg, prefix_handler.compile_nginx_conf, debug) end
+      local kong_ngx_cfg = function(cfg, debug) return _compile(cfg, prefix_handler.compile_kong_conf, debug) end
+
+      local debug = false
+      it("has no wasm{} block by default", function()
+        assert.not_matches("wasm {", ngx_cfg({ wasm = false }, debug))
```

Comment:
Shouldn't this be `{}` (or maybe `{ wasm = nil }` for clarity) if it's supposed test the default behavior? The wasm setting is not set/altered in `spec/kong_tests.conf`, so I think leaving it out should still give us the default behavior.